apps/test/unit/lib/kits/maker/CircuitPlaygroundBoardTest.js

```diff
@@ -0,0 +1,220 @@
+import sinon from 'sinon';
+import {expect} from '../../../../util/configuredChai';
+import {EventEmitter} from 'events'; // see node-libs-browser
+import Playground from 'playground-io';
+import CircuitPlaygroundBoard from '@cdo/apps/lib/kits/maker/CircuitPlaygroundBoard';
+
+// Polyfilll node process.hrtime for the browser, which gets used by johnny-five
+process.hrtime = require('browser-process-hrtime');
+
+describe('CircuitPlaygroundBoard', () => {
+  let board, playground;
+
+  beforeEach(() => {
+    // We use real playground-io, but our test configuration swaps in mock-firmata
+    // for real firmata (see webpack.js) changing Playground's parent class.
+    sinon.stub(CircuitPlaygroundBoard, 'makePlaygroundTransport', () => {
+      playground = new Playground({});
+      playground.SERIAL_PORT_IDs.DEFAULT = 0x08;
+
+      // mock-firmata doesn't implement these (yet) - and we want to monitor how
+      // they are called.
+      playground.sysexCommand = sinon.spy();
+      playground.sysexResponse = sinon.spy();
+
+      // Also spy on these
+      sinon.spy(playground, 'reset');
+      sinon.spy(playground, 'pinMode');
+      sinon.spy(playground, 'digitalWrite');
+      sinon.spy(playground, 'digitalRead');
+      sinon.spy(playground, 'analogWrite');
+      sinon.spy(playground, 'analogRead');
+
+      // Pretend to be totally ready
+      playground.emit('connect');
+      playground.emit('ready');
+
+      return playground;
+    });
+
+    // Construct a board to test on
+    board = new CircuitPlaygroundBoard();
+  });
+
+  afterEach(() => {
+    CircuitPlaygroundBoard.makePlaygroundTransport.restore();
+  });
+
+  it('is an EventEmitter', () => {
+    expect(board).to.be.an.instanceOf(EventEmitter);
+  });
+
+  describe(`connect()`, () => {
+    it('returns a Promise that resolves when the board is ready to use', () => {
+      return board.connect();
```

Comment:
Okay, I've removed this test (since our mock serial port doesn't test the "when ready to use" part and the promise part is asserted over and over in subsequent tests), and I've expanded the next test to contain a verbose list of components created on connect.  I'm not putting stronger assertions about what those components _are_ here because there's a separate set of tests for that in [PlaygroundComponentsTest.js](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/test/unit/lib/kits/maker/PlaygroundComponentsTest.js).